test/parallel/test-repl-use-global.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+// Flags: --expose-internals
+
+const common = require('../common');
+const stream = require('stream');
+const repl = require('internal/repl');
+const assert = require('assert');
+
+common.globalCheck = false;
+
+// Array of [useGlobal, expectedResult] pairs
+const globalTestCases = [
+  [false, 'undefined'],
+  [true, '\'tacos\''],
+  [undefined, 'undefined']
+];
+
+const globalTest = (useGlobal, cb, output) => (err, repl) => {
+  if (err)
+    return cb(err);
+
+  let str = '';
+  output.on('data', (data) => (str += data));
+  global.lunch = 'tacos';
+  repl.write('global.lunch;\n');
+  repl.close();
+  delete global.lunch;
+  cb(null, str.trim());
+};
+
+// Test how the global object behaves in each state for useGlobal
+for (const [option, expected] of globalTestCases) {
+  runRepl(option, globalTest, common.mustCall((err, output) => {
+    assert.ifError(err);
+    assert.strictEqual(output, expected);
+  }));
+}
+
+// Test how shadowing the process object via `let`
+// behaves in each useGlobal state. Note: we can't
+// actually test the state when useGlobal is true,
+// because the exception that's generated is caught
+// (see below), but errors are printed, and the test
+// suite is aware of it, causing a failure to be flagged.
+//
+const processTestCases = [false, undefined];
+const processTest = (useGlobal, cb) => (err, repl) => {
+  if (err)
+    return cb(err);
+  // if useGlobal is false, then `let process` should work
+  repl.write('let process;\n');
```

Comment:
nit: It might be nice to have a statement with output executed after this, e.g. `21 * 2` or so. You could then re-use the existing output handling for the `globalTest` cases + you’d be checking that the REPL still works.